components/script/dom/xmlhttprequest.rs

```diff
@@ -725,7 +728,18 @@ impl XMLHttpRequestMethods for XMLHttpRequest {
                         JS_ClearPendingException(cx);
                         return NullValue();
                     }
-                }
+                },
+                XMLHttpRequestResponseType::Blob => {
+                    // Step 1
```

Comment:
This step should also be moved to `blob_response`. All we should have here is a single call to `blob_response`.